### PR TITLE
reproc: new recipe

### LINKED ISSUE #978: [request] reproc and reprocxx
### Package Details
  * Package Name/Version: **reproc**
  * Website: **https://github.com/DaanDeMeyer/reproc**
  * Source code: **https://github.com/DaanDeMeyer/reproc**

### Description Of The Library / Tool

Process managing library

### PR DESCRIPTION
**Pull request by valgur**
### Summary
Adds  **reproc/14.2.5**

#### Motivation
https://github.com/DaanDeMeyer/reproc

reproc (Redirected Process) is a cross-platform C/C++ library that simplifies starting, stopping and communicating with external programs. The main use case is executing command line applications directly from C or C++ code and retrieving their output.

[Packaging status](https://repology.org/project/reproc/versions)

#### Details
This project exports two CMake config files (`reproc` and `reproc++`) but due to a Conan limitation the recipe exports just a single `reproc` one.

- Closes: https://github.com/conan-io/conan-center-index/issues/978
- Closes: https://github.com/conan-io/conan-center-index/issues/25531

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by AbrilRBS — COMMENTED** [review]
Thanks! Will check with the team about the cmake file being different in cxx mode, see if there's something that can be done :)

File: recipes/reproc/all/conanfile.py [hunk 1]
```diff
@@ -82,22 +82,21 @@ def package(self):
...
-        # The C++ component should have its separate reproc++-config.cmake file,
-        # but it's currently not supported by Conan
-        self.cpp_info.set_property("cmake_file_name", "reproc")
+        cmake_config_name = "reproc++" if self.options.cxx else "reproc"
+        self.cpp_info.set_property("cmake_file_name", cmake_config_name)
 
-        self.cpp_info.components["reproc_c"].set_property("pkg_config_name", "reproc")
-        self.cpp_info.components["reproc_c"].set_property("cmake_target_name", "reproc")
-        self.cpp_info.components["reproc_c"].libs = ["reproc"]
+        self.cpp_info.components["reproc"].set_property("pkg_config_name", "reproc")
```
Comment (valgur) [inline recipes/reproc/all/conanfile.py:88]: @czoido This makes the `reproc::reproc` Conan component ambiguous. I'm not even sure whether `cpp_info.requires = ["reproc::reproc"]` and `CMakeDeps.set_property("reproc::reproc", ...)` would interpret this as the aggregate component or the individual one.

**Review by franramirez688 — DISMISSED** [review]

File: recipes/reproc/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            self.options.rm_safe("fPIC")
+        if not self.options.cxx:
+            self.settings.rm_safe("compiler.cppstd")
+            self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.options.cxx and self.settings.compiler.cppstd:
```
Comment (AbrilRBS) [inline recipes/reproc/all/conanfile.py:49]:
```suggestion
        if self.options.cxx:
```

if cppstd is not set for cxx, then something is broken, break

**Review by AbrilRBS — APPROVED** [review]
Looks great, thanks!

**Review by franramirez688 — APPROVED** [review]

**Pull request merged**